app/models/imports/doc.rb

```
@@ -2,5 +2,28 @@ module Imports
   class Doc < Import
     has_one :pdf, as: :parent, dependent: :destroy, autosave: true
     has_one_attached :file
+
+    def process
+      output_pdf_path = ConvertDocToPdf.call(id, file)
+
+      transaction do
+        create_pdf!(
+          file: File.open(output_pdf_path),
+          public_document: public_document,
+          metadata: metadata,
+        )
+        update!(
+          processed_at: Time.now,
+          processing_errors: nil,
+          status: :completed,
+        )
+      end
+    rescue ConvertDocToPdf::PdfConversionError => e
```

We should probably catch all exceptions, log them, and then re-raise them.